### DISCUSSION TITLE
Elsa v3 Entirely unreliable with many long-running workflows, I need help configuring it for better performance.

### DISCUSSION THREAD

**Discussion by kyle-alm** [elsa-workflows/elsa-core#5880] [Q&A]
I'm using Elsa v3 for a few thousand long running workflows. These are customer contact workflows. Send an email, wait three days, etc.

I have 4884 running right now, and I've only just started, but to say these are running is charitable. They're nearly all locked up. They're resuming hours after they're supposed to, and we've scaled the server with enough cpu to handle it all. Every request to elsa takes 10 seconds, even if it goes straight to a 1 second delay (that actually takes an hour to resume) for "immediate" http response. Some of these are suspended but not showing that they actually are on a delay, just between two activities.

When I look for information, I mostly find stuff on Elsa v2. I know there are ways to configure things, but I can't find information on that at all. Elsa v2 you could change the sweep interval? No documentation on that for v3 that I can find. Any help would be appreciated!

**Comment by ArkComet** [comment]
Were you able to find a solution? Is this issue from running delayed workflows to that large of a scale? I was planning on a use case that is much smaller, about 10-25 workflows.

**Reply by kyle-alm** [reply]
My solution was to build my own system and migrate off of Elsa. I'm sure 10-25 workflows would do fine, but it's incredibly inefficient. Elsa stores mission-critical data like when an activity will execute in json in the database, and does a poor job parsing that data. For example, we found that if you provide dates in ISO 8601 format (ending in a 'z'), elsa just doesn't parse them. No errors, you'll just find dates come and go and nothing happens. I would personally avoid Elsa in it's current state (and in particular it's current state of documentation).
